gc/base/standard/ConcurrentGC.cpp

```diff
@@ -2207,7 +2196,7 @@ MM_ConcurrentGC::concurrentMark(MM_EnvironmentStandard *env, MM_MemorySubSpace *
 			if(_stats->switchExecutionMode(CONCURRENT_ROOT_TRACING1, CONCURRENT_ROOT_TRACING2)) {
 				/* Signal threads for async callback */
 				bool signaledThreadsForAsyncCallback = false;
-				_cli->concurrentGC_collectRoots(env, CONCURRENT_ROOT_TRACING1, &_scanClassesMode, signaledThreadsForAsyncCallback, taxPaid);
```

Comment:
Looking into this. Stayed away from it because concurrent GC owns execution mode and drives state transitions. ROOT_TRACING* states are sequential so I can probably collapse them into a single ROOT_TRACING state where concurrent GC uses an alternate protocol to handshake with client language to manage buffering and helper threads while tracing roots in payAllocationTax().